hail/python/hail/expr/generic_summary.py

```diff
@@ -0,0 +1,108 @@
+import hail as hl
+
+
+def pct(x):
+    return f'{x*100:.2f}%'
+
+
+class Computations(object):
+    def __init__(self):
+        super(Computations, self).__init__()
+        self.n = 0
+        self.l = []
+
+    def append(self, agg):
+        self.l.append(agg)
+        n_ret = self.n
+        self.n += 1
+        return n_ret
+
+    def result(self):
+        return hl.tuple(self.l)
+
+
+def format(x):
+    if isinstance(x, float):
+        return f'{x:.2f}'
+    else:
+        return x
+
+
+def map_int(x):
+    if x is None:
+        return x
+    else:
+        return int(x)
+
+
+def generic_summary(x, prefix=''):
+    computations = Computations()
+    to_print = []
+
+    def append_agg(agg):
+        return computations.append(agg)
+
+    count = append_agg(hl.agg.count())
+    to_print.append(('(Summary)', {'Number of records': lambda results: format(results[count])}))
+
+    def recur_expr(expr, path):
+        d = {}
+        missingness = append_agg(hl.agg.count_where(hl.is_missing(expr)))
+        d['missing'] = lambda \
+                results: f'{results[missingness]} values ({pct(results[missingness] / results[count])})'
+
+        t = expr.dtype
+
+        if t in (hl.tint32, hl.tint64, hl.tfloat32, hl.tfloat64):
+            stats = append_agg(hl.agg.stats(expr))
+            if t in (hl.tint32, hl.tint64):
+                d['minimum'] = lambda results: format(map_int(results[stats]['min']))
+                d['maximum'] = lambda results: format(map_int(results[stats]['max']))
+            else:
+                d['minimum'] = lambda results: format(results[stats]['min'])
+                d['maximum'] = lambda results: format(results[stats]['max'])
+            d['mean'] = lambda results: format(results[stats]['mean'])
+            d['stdev'] = lambda results: format(results[stats]['stdev'])
+            d['sum'] = lambda results: format(results[stats]['sum'])
```

Comment:
Should sum be displayed as an integer if the field type is?

Reply:
good call.